qiskit/algorithms/minimum_eigen_solvers/adapt_vqe.py

```diff
@@ -0,0 +1,338 @@
+    UNKNOWN = "The algorithm finished due to an unforeseen reason"
+
+
+class AdaptVQE(VariationalAlgorithm):
+    """The Adaptive Variational Quantum Eigensolver algorithm.
+
+    `AdaptVQE <https://arxiv.org/abs/1812.11173>`__ is a quantum algorithm which creates a compact
+    ansatz from a set of evolution operators. It iteratively appends excitations with the largest energy
+    gradient to the circuit. This results in a wavefunction ansatz which is uniquely adapted to the operator
+    whose minimum eigenvalue is being determined.
+    This class relies internally on a `VQE` to find the minimum eigenvalue
+    of the provided operator.
+    The performance of AdaptVQE can significantly depend on the choice of gradient method, QFI
+    solver (if applicable) and the epsilon value.
+    """
+
+    def __init__(
+        self,
+        solver: VQE,
+        adapt_gradient: Optional[GradientBase] = None,
+        excitation_pool: List[Union[OperatorBase, QuantumCircuit]] = None,
+        threshold: float = 1e-5,
+        max_iterations: Optional[int] = None,
+    ) -> None:
+        """
+        Args:
+            solver: a `VQE` instance used internally to compute the minimum eigenvalues.
+               It is a requirement that the `ansatz` of this solver is of type `EvolvedOperatorAnsatz`.
+            adapt_gradient: a `Gradient` instance with which to compute the excitation operator gradients.
+                If `None`, this will default to a parameter shift gradient.
+            excitation_pool: A list of quantum circuits out of which to build the ansatz.
+            threshold: the eigenvalue convergence threshold. It has a minimum value of `1e-15`.
+            max_iterations: the maximum number of iterations.
+        """
+        validate_min("threshold", threshold, 1e-15)
+
+        if adapt_gradient is None:
+            adapt_gradient = Gradient(grad_method="param_shift")
+        self._threshold = threshold
+        self.solver = solver
+        self._tmp_ansatz = self.solver.ansatz
+        self._max_iterations = max_iterations
+        self._adapt_gradient = adapt_gradient
+        self._excitation_pool = excitation_pool
+        self._excitation_list: List[OperatorBase] = []
+
+    @property
+    def initial_point(self) -> Optional[np.ndarray]:
+        """Returns the initial point."""
+        return self.solver.initial_point
+
+    @initial_point.setter
+    def initial_point(self, initial_point: Optional[np.ndarray]) -> None:
+        """Sets the initial point."""
+        self.solver.initial_point = initial_point
+
+    def _compute_gradients(
+        self,
+        theta: List[float],
+        operator: OperatorBase,
+    ) -> List[Tuple[float, PauliSumOp]]:
+        """
+        Computes the gradients for all available excitation operators.
+
+        Args:
+            theta: List of (up to now) optimal parameters
+            operator: operator whose gradient needs to be computed
```

Comment:
Let's make it consistent across the whole code that we start with a capital letter after `:` and finish with `.`.